toponetx/classes/path.py

```diff
@@ -0,0 +1,162 @@
+        Else, the sub-sequence of nodes in the p-path will be reversed if the first index is larger than the last index.
+    allowed_paths : Iterable[tuple[Hashable]], optional
+        A list of allowed boundaries. If None, all possible boundaries are constructed (similarly to simplex).
+    attr: keyword arguments, optional
+        Additional attributes to be associated with the p-path.
+    """
+
+    def __init__(
+        self,
+        elements: Sequence[Hashable],
+        name: str = "",
+        construct_boundaries: bool = False,
+        reserve_sequence_order: bool = False,
+        allowed_paths: Iterable[tuple[Hashable]] = None,
+        **attr,
+    ) -> None:
+        self.__check_inputs(elements, reserve_sequence_order)
+        super().__init__(tuple(elements), name, **attr)
+        if len(set(elements)) != len(self.elements):
+            raise ValueError("A p-path cannot contain duplicate nodes.")
+
+        self.construct_boundaries = construct_boundaries
+        self.reserve_sequence_order = reserve_sequence_order
+        if construct_boundaries:
+            self._boundaries = self.construct_path_boundaries(
+                elements,
+                reserve_sequence_order=reserve_sequence_order,
+                allowed_paths=allowed_paths,
+            )
+        else:
+            self._boundaries = list()
+
+    @staticmethod
+    def construct_path_boundaries(
+        elements: Sequence[Hashable],
+        reserve_sequence_order: bool = False,
+        allowed_paths: Iterable[tuple[Hashable]] = None,
+    ) -> list[tuple[Hashable]]:
+        """
+        Return list of p-path objects representing the boundaries.
+
+        Parameters
+        ----------
+        elements: Sequence[Hashable]
+            The nodes in the p-path.
+        reserve_sequence_order : bool, default=False
+            If True, reserve the order of the sub-sequence of nodes in the p-path.
+            Else, the sub-sequence of nodes in the p-path will be reversed if the first index is larger than the last index.
+        allowed_paths : Iterable[tuple[Hashable]], optional
+            A list of allowed boundaries. If None, all possible boundaries are constructed (similarly to simplex).
+
+        Returns
+        -------
+        boundaries : list[tuple[Hashable]]
+            List of p-path objects representing the boundaries.
+        """
+        boundaries = list()
+        for i in range(len(elements)):
+            boundary = list(elements[0:i] + elements[(i + 1) :])
+            if (
+                not reserve_sequence_order
+                and len(boundary) > 1
+                and boundary[0] > boundary[-1]
+            ):
+                boundary.reverse()
+            if allowed_paths is None or tuple(boundary) in allowed_paths:
+                boundaries.append(tuple(boundary))
+        return boundaries
+
+    @property
+    def boundary(self) -> list[tuple[Hashable]]:
```

Comment:
define what you mean by boundary here. Note, the term boudary matrix is not used in toponetx, instead we are using incidence matrix. If this is what you mean then you probably need to change it to be consistent

Reply:
Boundary here conveys the same meaning with the boundary method in `simplex.py` and `cell.py`. I just want `path.py` to follow the similar interface.

Reply:
sure, we should define that in these classes as well. Please raise an issue on that.

Reply:
I mean `boundary()` already exists in `simplex.py` and `cell.py`, so no need to raise an issue on this.

https://github.com/pyt-team/TopoNetX/blob/da155dfbb46a77a9034a1d5707b573f16b93cce9/toponetx/classes/simplex.py#L100-L107

https://github.com/pyt-team/TopoNetX/blob/da155dfbb46a77a9034a1d5707b573f16b93cce9/toponetx/classes/cell.py#L205-L215